src/main/java/io/cdap/e2e/pages/locators/WranglerLocators.java

```diff
@@ -0,0 +1,149 @@
+  public static WebElement enterText;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-findAndReplace-oldValueInput']")
+  public static WebElement enterOldValue;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-encodeDecode-title-Encode']")
+  public static WebElement encodeDirective;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-encodeDecode-title-Decode']")
+  public static WebElement decodeDirective;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-findAndReplace-newValueInput']")
+  public static WebElement enterNewValue;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(@class, 'level-popover')]//textarea")
+  public static WebElement filterTextArea;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(@class, 'level-popover')]//select")
+  public static WebElement filterSelect;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-singleField-" +
+    "mainFieldInput']")
+  public static WebElement columnWidths;
+
+  @FindBy(how = How.XPATH, using = "//select[@data-testid='features-dataprep-directives-defineVariable-" +
+    "columnSelector']")
+  public static WebElement selectColumn;
+
+  @FindBy(how = How.XPATH, using = "//select[@data-testid='features-dataprep-directives-defineVariable-" +
+    "conditionSelector']")
+  public static WebElement selectRow;
+
+  @FindBy(how = How.XPATH, using = "//input[@placeholder='Enter depth']")
+  public static WebElement enterDepth;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-dateFormats-" +
+    "customFormatInput']")
+  public static WebElement customDate;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-setCounter-counterNameInput']")
+  public static WebElement enterCounterName;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-setCounter-" +
+    "counterIncrementInput']")
+  public static WebElement incrementCount;
+
+  public static WebElement locateTransformationButton(String columnName) {
+    return SeleniumDriver.getDriver()
+      .findElement(By.xpath("//th[@id='column-" + columnName + "']//button[@data-testid='features-dataprep-workspace" +
+                              "-dataTable-head-directivesDropdown-toggleButton']"));
+  }
+
+  public static WebElement locateDirectivesTitle(String directiveName) {
+    return SeleniumDriver.getDriver()
+      .findElement(By.xpath("//div[@data-testid='features-dataprep-" +
+                              "directives-" + directiveName + "-title']"));
+  }
+
+  public static WebElement locateCheckboxOfColumn(String columnName) {
+    return SeleniumDriver.getDriver()
+      .findElement(By.xpath("//th[@id='column-" + columnName + "']/descendant::span[@data-testid='" +
```

Comment:
Can be located uniquely using testids only. Please use the chrome extension to get the XPATH.

Reply:
Is the //th[@id= ... ] necessary ? Can the column header not be identified using data-testid ?